File: Project_WEB_AMS/WebContent/js/Account.js
/**
 *  계좌 정보를 저장하는 Account 생성자
 */

// 생성자
function Account(num,owner,pw,rest) {
	this.num = num;
	this.owner = owner;
	this.pw = pw;
	this.rest = rest;
}

// Account의 모든 인스턴스가 공유하는 변수
Account.bankName = "KOSTA 은행";

// Account의 prototype에 메소드 선언
// 입금시 money만큼 잔액이 증가하는 메소드
Account.prototype.deposit = function(money) {
	if(money <= 0) return false;
	this.rest += money;
	return this.rest;
}
// 출금시 money만큼 잔액이 감소하는 메소드
Account.prototype.withdraw = function(money) {
	if(money <= 0) return false;
	if(money > this.rest) return false;
	this.rest -= money;
	return this.rest;
}
// 패스워드를 검사하는 메소드
Account.prototype.checkPasswd = function (pw) {
	return this.pw == pw;
}
// 계좌 정보를 문자열로 반환하는 메소드
Account.prototype.toString = function() {
	return Account.bankName + " " + this.num + " " + this.owner + " " + this.pw + " " + this.rest;
}

File: Project_MyBlog_Teacher/src/kr/or/kosta/jsp/Account.java
package kr.or.kosta.jsp;

/**
 * JavaBean
 */
public class Account {
	
	public  static final String bankName = "하나은행";

	private String accountNum ;
	private String accountOwner;
	private int passwd;
	private long restMoney;

	public Account(){
		this(null, null);
	}

	public Account(String accountNum, String accountOwner){
		//this.accountNum = accountNum;
		//this.accountOwner = accountOwner;
		this(accountNum, accountOwner, 0, 0);
	}

	public Account(String accountNum, String accountOwner, int passwd, long restMoney){
		this.accountNum = accountNum;
		this.accountOwner = accountOwner;
		this.passwd = passwd;
		this.restMoney = restMoney;
	}

	public void setAccountNum(String accountNum){
		this.accountNum = accountNum;
	}

	public String getAccountNum(){
		return accountNum;	
	}

	public void setAccountOwner(String accountOwner){
		this.accountOwner = accountOwner;
	}

	public String getAccountOwner(){
		return accountOwner;	
	}

	public void setRestMoney(long restMoney){
		this.restMoney = restMoney;
	}

	public void setPasswd(int passwd){
		this.passwd = passwd;
	}

	public int getPasswd(){
		return passwd;
	}


	// 인스턴스 메소드
	public long deposit(long money){
		restMoney += money;
		return restMoney;
	}

	public long withdraw(long money){
		restMoney -= money;
		return restMoney;
	}

	public long getRestMoney(){
		return restMoney;
	}

	public boolean checkPasswd(int pw){
		return passwd == pw;
	}
	
	@Override
	public String toString() {
		return getAccountNum() + "\t" + getAccountOwner() + "\t" + "****" + "\t" + getRestMoney();
	}
	
	@Override
	public boolean equals(Object obj) {
		boolean eq = false;
		if(obj instanceof Account) {
			eq = toString().equals(obj.toString());
		}
		return eq;
	}
}
